### DISCUSSION TITLE
Modelling multiple frequencies: subsequent or all-parameter fitting?

### DISCUSSION THREAD

**Discussion by LRBaalmann** [giordano/LombScargle.jl#34] [Q&A]
Hi all,

I greatly appreciate the many functions of this package. However, I'm a bit concerned whether the subsequent fitting of frequencies with the model function, as is suggested in the documentation, is the best way to go:

The standard approach when fitting multiple oscillations with DFT on evenly spaced data is to perform a least-squares fit with the amplitudes and phases of all involved frequencies as fitting parameters; doing so subsequently yields incorrect results. (I'm sure there's a canonical reference for this that I'm not aware of, but it's at least mentioned in Kurtz1985, 1985MNRAS.213..773K, and employed as such by, for example, the program period04.)

The LombScargle.model function works completely differently, so perhaps my concerns are uncalled for, and subsequent fitting does not introduce the problems it does for Fourier. Nevertheless, I thought I should mention it.

Thanks!

**Comment by giordano** [comment]
Uhm, this is an interesting question I do not have a definitive answer for.  I also don't have much time to investigate this, to be honest, I'd be curious to hear if you can discover more.
